solar_system.rb

```diff
@@ -0,0 +1,92 @@
+#WAVE 3 BASIC
+
+class Planet
+  attr_reader :names, :diameter, :mass, :number_moons, :color
+
+  def initialize(name, diameter, mass, number_moons, color)
+    @name = name
+    @diameter = diameter
+    @mass = mass
+    @number_moons = number_moons
+    @color = color
+  end
+
+  def name
+    return @name
+  end
+
+  def display_planets
+    return "name: #{@name}, diameter: #{@diameter}, mass: #{@mass}, number of moons: #{@number_moons}, color: #{@color} \n"
+  end
+end
+
+class Solar_system
+  attr_reader :planets
+
+  def initialize(planets)
+    @planets = planets
+  end
+
+  def add_planet(planet)
+    @planets << planet
+  end
+
+  def list_planets
+    x = 0
+    @planet_names = ""
+    @planets.each do |planet|
+      x+=1
+      @planet_names << "#{x}. #{planet.name} \n"
+    end
+    return @planet_names
+  end
+
+  def make_planet_array
+    @planet_array = []
+    @planets.each do |planet|
+      @planet_array << "#{planet.display_planets}"
+    end
+    return @planet_array
+  end
+
+  def show_planets(response)
+    @response = response.to_i - 1
+    if @response > @planet_array.length
+      return "Please select a number displayed on the menu, idiot"
+    end
+    return @planet_array[@response]
+  end
+end
+
+
+venus = Planet.new("Venus", 20, 45, 5, "red")
+saturn = Planet.new("Saturn", 43, 23, 1, "yellow")
+pluto = Planet.new("Pluto", 12, 21, 0, "blue")
+
+milky_way = Solar_system.new([venus, saturn, pluto])
```

Comment:
Since you're creating all these local variables just to add them to an array, a more concise way to do this might be:

```ruby
planet_array = [
  Planet.new("Venus", 20, 45, 5, "red"),
  Planet.new("Saturn", 43, 23, 1, "yellow"),
  Planet.new("Pluto", 12, 21, 0, "blue")
]
solar_system = SolarSystem.new(planet_array)
```

This makes it clear to the reader that we won't be using the planets individually, only through the collection, and also saves you having to come up with a bunch of variable names.